### PR TITLE
chore: update procedure map tests to match the refactored API

### LINKED ISSUE #6561: Update procedure map tests to match the refactored API
### Description

The API designed in the [design doc](https://docs.google.com/document/d/1n4MxZQu49mniC7aUtZ1X4xvwWgpL3st4aDU25pwdaNc/edit?usp=sharing) was refactored in response to feedback on #6488 and #6489. The tests written before these (in  #6486) need to be updated to reflect the new design.

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [ ] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

### Resolves

Fixes #6561 

### Proposed Changes

🎵 Updates the tests 🎵

### Reason for Changes

Tests should test.

### Test Coverage

This is tests silly!

### Documentation

N/A

### Additional Information

Dependent on #6560

### CODE REVIEW

File: tests/mocha/procedure_map_test.js [hunk 1]
```diff
@@ -36,20 +36,28 @@ suite.skip('Procedure Map', function() {
...
 
       test('deleting a procedure triggers an update', function() {
-        const procedureModel =
-            new Blockly.procedures.ObservableProcedureModel(this.workspace);
-        this.procedureMap.addProcedure(procedureModel);
+        this.procedureMap.add(
+            new Blockly.procedures.ObservableProcedureModel(this.workspace));
 
-        this.procedureMap.deleteProcedure(procedureModel.getId());
+        this.procedureMap.delete([...this.procedureMap.values()][0].getId());
```
Comment (maribethb) [inline tests/mocha/procedure_map_test.js:60]: why this change? the old way is more readable imo (here and below)

(i know the method name changed, but storing the procedureModel in a variable)

File: tests/mocha/procedure_map_test.js [hunk 2]
```diff
@@ -8,18 +8,18 @@ import {sharedTestSetup, sharedTestTeardown} from './test_helpers/setup_teardown
...
-    this.procedureMap = this.workspace.getProcedureMap();
+    // this.procedureMap = this.workspace.getProcedureMap();
   });
 
   teardown(function() {
     sharedTestTeardown.call(this);
   });
 
-  suite('triggering block updates', function() {
+  suite.skip('triggering block updates', function() {
```
Comment (maribethb) [inline tests/mocha/procedure_map_test.js:22]: just checking this is supposed to be skipped now? can you add a TODO comment saying when to unskip it?

**Review by maribethb — APPROVED** [review]

File: tests/mocha/procedure_map_test.js (hunk 1 above)
Comment (BeksOmega) [reply tests/mocha/procedure_map_test.js:60]: I was moving the tests to actually use the fluent API hehe. I'll move things around so that we're storing the relevant bit in a variable though.

(side question: Do you think it'd be more readable as `this.procedureMap.getProcedures()[0].getId()`? )

File: tests/mocha/procedure_map_test.js (hunk 2 above)
Comment (BeksOmega) [reply tests/mocha/procedure_map_test.js:22]: Yesh! The skip just got moved from the top-level down to here. Can add a comment o7

File: tests/mocha/procedure_map_test.js [hunk 3]
```diff
@@ -184,4 +191,67 @@ suite.skip('Procedure Map', function() {
...
+
+  suite('backing variable to parameters', function() {
+    test(
+        'construction references an existing variable if available',
+        function() {
+          const variable = this.workspace.createVariable('test1');
+          const param = new Blockly.procedures.ObservableParameterModel(
+              this.workspace, 'test1');
+
+          console.log(param);
```
[the remark this answers is not in the document]
Comment (BeksOmega) [reply tests/mocha/procedure_map_test.js:203]: Thanks for the catch!

File: tests/mocha/procedure_map_test.js (hunk 1 above)
Comment (maribethb) [reply tests/mocha/procedure_map_test.js]: yeah, that would be more readable, but i think storing it in a variable is best.

you can have a separate test that tests the getProcedures function and one that tests the chain of the fluent api.

the justification is that if one of those methods break, you ideally want only the tests related to it to break, and not every single one of the tests, because that would slow down finding out the actual cause of the breakage. 

I think these are relevant snippets from go/readable-tests:
- "Create local variables when nontrivial values are referenced multiple times"
- "Write tests as straight-line code that doesn't force the reader to do mental computations to verify the test's correctness"
- "Write a separate test for each behavior defined by a method"
- "Write tests that call multiple methods on a class only when that sequence of calls defines a well-specified behavior"

Comment (BeksOmega) [reply tests/mocha/procedure_map_test.js]: Do you think it's useful to test the chain of the fluent API? Imo I think that's well specified by the interfaces, and additional tests would just end up being change detectors. But idk, I'm still learning how to write tests well. (also acknowledging that you could have just been using that as an example haha)

I just wanted to change these so that they matched how we want people to actually use the API, not necessarily /test/ the API if that makes sense.

And wrt your snippets, were there any other changes you wanted me to make?

Comment (BeksOmega) [reply tests/mocha/procedure_map_test.js]: > but in terms of how people would use it, i think it's important that you stick to public APIs, but not important that everything you do in the test would necessarily make sense to do in production, if that makes sense.

Sweet, totally makes sense!
